### PR TITLE
Keep Spigot in its own submodule

### PR DESCRIPTION
**Pull request by Techcable**
Prevents us from having to merge spigot when it changes, instead we just update submodules.
This seperates our history and better organizes the project.

Based on TacoSpigot/TacoSpigot@cd6bc2fab25eedf767e91145795e188c5a7d2170

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -10,3 +10,6 @@
 [submodule "Paperclip"]
 	path = Paperclip
 	url = https://github.com/PaperSpigot/Paperclip.git
+[submodule "Spigot"]
+	path = Spigot
+	url = https://Techcable@hub.spigotmc.org/stash/scm/spigot/spigot.git
```
Comment (Black-Hole) [inline .gitmodules:15]: :wave:

**Comment by Techcable** [comment]
Every day this stays open without being pulled is another commit I have to rebase with.
This is tested to work, and I already use it in tacospigot.

**Comment by kashike** [comment]
@Techcable See @Black-Hole's comment.

**Comment by Techcable** [comment]
oooh. I thought he was clapping from being happy, not sarcasim.

**Comment by Ichbinjoe** [comment]
I vote keep it open to make Techcable mad :P

**Comment by zachbr** [comment]
Then don't bother updating it, this isn't a PR that needs a big green happy merge button. I am more than happy to merge it manually when I have some time to make sure it works (which looks fine) and works well with our CI system.

**Comment by zachbr** [comment]
Merged w/ 667953bb9f67ede3bef1843fc39c3e3183c51861

**Pull request closed without merging**